reproducibility_project/src/analysis/sampler.py

```diff
@@ -0,0 +1,57 @@
+"""Use the pymbar package to perform decorrelated equilibration sampling."""
+
+import numpy as np
+from pymbar import timeseries
+
+from reproducibility_project.src.analysis.equlibration import is_equilibrated
+
+
+def sample_job(job, variable="potential_energy", threshold=0.75):
+    """Use the timeseries module from pymbar to perform statistical sampling.
+
+    The start, end and decorrleated step size of the production region are
+    added to the job document.
+
+    Parameters
+    ----------
+    job : signac.contrib.job.Job
+        The Job object.
+    variable : str; default "potential_energy"
+        The variable to be used in sampling.
+    threshold : float, optional, default=0.75
+        Fraction of data expected to be equilibrated.
+    """
+    data = np.genfromtxt(job.fn("log.txt"), names=True)[variable]
+    start, stop, step = _decorr_sampling(data, threshold)
+    job.doc["sampled_variable"] = variable
```

Comment:
If we want to measure two properties like "density" and "potential energy", I dont think this would be able to store the `sample_start`, etc. for multiple properties. We might need to adjust the variable names or maybe write out a separate mini json files for each property.

Reply:
If we want to sample multiple variables and store the params for each, maybe a dictionary would work?

```
sample_results = {}
sample_results[variable] = range(start, stop, step)

job.doc["sample_results"] = sample_results:
```

Reply:
@justinGilmer I just pushed a commit along the lines of my last comment, let me know what you think of this approach.